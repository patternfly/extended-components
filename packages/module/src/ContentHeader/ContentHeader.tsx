import React from 'react';
import {
  Flex,
  FlexItem,
  Split,
  SplitItem,
  Text,
  PageSection,
  TextContent,
  Button,
  ButtonVariant,
  ButtonProps,
  Divider,
} from '@patternfly/react-core';
import { ExternalLinkAltIcon } from '@patternfly/react-icons';
import { createUseStyles } from 'react-jss';

export interface PageHeaderLinkProps extends ButtonProps {
  /** Title for the link */
  label: string;
  /** Indicates if the link points to an external page */
  isExternal?: boolean;
}

export interface ContentHeaderProps extends React.PropsWithChildren {
  /** Title for content header */
  title: string;
  /** Optional subtitle for content header */
  subtitle?: string;
  /** Optional link below subtitle */
  linkProps?: PageHeaderLinkProps;
  /** Optional icon for content header (appears to the left of the content header's title with a divider) */
  icon?: React.ReactNode;
  /** Optional label for content header (appears to the right of the content header's title) */
  label?: React.ReactNode;
  /** Breadcrumbs component */
  breadcrumbs?: React.ReactNode;
  /** Menu that appears to the far right of the title */
  actionMenu?: React.ReactNode;
  /** Custom OUIA ID */
  ouiaId?: string | number;
  /** Child nodes */
  children?: React.ReactNode;
}

const useStyles = createUseStyles({
  iconMinWidth: {
    minWidth: '48px',
  }
});

export const ContentHeader: React.FunctionComponent<React.PropsWithChildren<ContentHeaderProps>> = ({
  title,
  subtitle,
  linkProps,
  icon,
  label,
  breadcrumbs = null,
  actionMenu,
  children = null,
  ouiaId = 'ContentHeader',
}: ContentHeaderProps) => {
  const classes = useStyles();

  return (
    <PageSection variant="light">
      { breadcrumbs && (
        <div className="pf-v5-u-mb-md">
          {breadcrumbs}
        </div>
      )}
      <Flex>
        {icon && (
          <>
            <FlexItem alignSelf={{ default: 'alignSelfCenter' }} className={`${classes.iconMinWidth}`}>
              {icon}
            </FlexItem>
            <Divider orientation={{
              default: 'vertical',
            }} />
          </>
        )}
        <FlexItem flex={{ default: 'flex_1' }}>
          <Split hasGutter>
            <SplitItem>
              <TextContent>
                <Text className="pf-v5-u-mb-sm" component="h1" ouiaId={`${ouiaId}-title`}>
                  {title}
                </Text>
              </TextContent>
            </SplitItem>
            {label && (
              <SplitItem>
                {label}
              </SplitItem>
            )}
            <SplitItem isFilled />
            {actionMenu && (
              <SplitItem>
                {actionMenu}
              </SplitItem>
            )}
          </Split>
          <TextContent>
            {subtitle && (
              <Text component="p" ouiaId={`${ouiaId}-subtitle`}>
                {subtitle}
              </Text>
            )}
            {linkProps && (
              <Button variant={ButtonVariant.link} component="a" ouiaId={`${ouiaId}-link-button`} isInline icon={linkProps.isExternal ? <ExternalLinkAltIcon className='pf-v5-u-ml-sm' /> : null} iconPosition="end" {...linkProps}>
                {linkProps.label}
              </Button>
            )}
          </TextContent>
        </FlexItem>
      </Flex>
      {children}
    </PageSection>
  )};

export default ContentHeader;
