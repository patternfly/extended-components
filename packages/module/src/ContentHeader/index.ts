export { default } from './ContentHeader'
export * from './ContentHeader'