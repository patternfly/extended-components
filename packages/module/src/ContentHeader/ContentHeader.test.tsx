import React from 'react';
import { render } from '@testing-library/react';
import ContentHeader from './ContentHeader';

describe('Contentheader component', () => {
  test('should render', () => {
    expect(render(<ContentHeader title='My title' subtitle='This is a subtitle for your content header' />)).toMatchSnapshot();
  });
});