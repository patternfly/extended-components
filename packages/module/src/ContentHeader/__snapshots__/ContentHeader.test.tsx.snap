// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contentheader component should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="pf-v6-c-page__main-section"
      >
        <div
          class="pf-v6-l-flex"
        >
          <div
            class="pf-m-flex-1"
          >
            <div
              class="pf-v6-l-split pf-m-gutter"
            >
              <div
                class="pf-v6-l-split__item"
              >
                <div
                  class="pf-v6-c-content"
                >
                  <h1
                    class="pf-v6-c-content--h1 pf-v6-u-mb-sm"
                    data-ouia-component-id="ContentHeader-title"
                    data-ouia-component-type="PF6/Text"
                    data-ouia-safe="true"
                    data-pf-content="true"
                  >
                    My title
                  </h1>
                </div>
              </div>
              <div
                class="pf-v6-l-split__item pf-m-fill"
              />
            </div>
            <div
              class="pf-v6-c-content"
            >
              <p
                class="pf-v6-c-content--p"
                data-ouia-component-id="ContentHeader-subtitle"
                data-ouia-component-type="PF6/Text"
                data-ouia-safe="true"
                data-pf-content="true"
              >
                This is a subtitle for your content header
              </p>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="pf-v6-c-page__main-section"
    >
      <div
        class="pf-v6-l-flex"
      >
        <div
          class="pf-m-flex-1"
        >
          <div
            class="pf-v6-l-split pf-m-gutter"
          >
            <div
              class="pf-v6-l-split__item"
            >
              <div
                class="pf-v6-c-content"
              >
                <h1
                  class="pf-v6-c-content--h1 pf-v6-u-mb-sm"
                  data-ouia-component-id="ContentHeader-title"
                  data-ouia-component-type="PF6/Text"
                  data-ouia-safe="true"
                  data-pf-content="true"
                >
                  My title
                </h1>
              </div>
            </div>
            <div
              class="pf-v6-l-split__item pf-m-fill"
            />
          </div>
          <div
            class="pf-v6-c-content"
          >
            <p
              class="pf-v6-c-content--p"
              data-ouia-component-id="ContentHeader-subtitle"
              data-ouia-component-type="PF6/Text"
              data-ouia-safe="true"
              data-pf-content="true"
            >
              This is a subtitle for your content header
            </p>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
